packages/core/database/lib/entity-manager/entity-repository.js

```diff
@@ -99,8 +108,38 @@ const createRepository = (uid, db) => {
       return db.entityManager.populate(uid, entity, populate);
     },
 
-    load(entity, field, params) {
-      return db.entityManager.load(uid, entity, field, params);
+    load(entity, fields, params) {
+      return db.entityManager.load(uid, entity, fields, params);
+    },
+
+    async loadPages(entity, field, params) {
+      if (!isString(field)) {
+        throw new Error(`Invalid load. Expected ${field} to be a string`);
+      }
+
+      const { attributes } = db.metadata.get(uid);
+      const attribute = attributes[field];
+
+      if (!attribute || attribute.type !== 'relation' || !isAnyToMany(attribute)) {
+        throw new Error(`Invalid load. Expected ${field} to be an anyToMany relational attribute`);
+      }
+
+      const [query, { page, pageSize }] = withOffsetLimit(params);
+
+      const [results, { count: total }] = await Promise.all([
+        db.entityManager.load(uid, entity, field, query),
+        db.entityManager.load(uid, entity, field, { ...query, count: true }),
+      ]);
+
+      return {
+        results,
+        pagination: {
+          page,
+          pageSize,
+          pageCount: Math.ceil(total / pageSize),
+          total,
+        },
+      };
```

Comment:
very clean! now the controller has much less boilerplate

Reply:
😊